ext/standard/php_fopen_wrapper.c

```diff
@@ -218,6 +218,16 @@ php_stream * php_stream_url_wrap_php(php_stream_wrapper *wrapper, const char *pa
 	}
 
 	if (!strcasecmp(path, "input")) {
+		path += 5;
+		max_memory = SAPI_POST_BLOCK_SIZE;
+		if (!strncasecmp(path, "/maxmemory:", 11)) {
+			path += 11;
+			max_memory = ZEND_STRTOL(path, NULL, 10);
+			if (max_memory < 0) {
+				zend_throw_error(NULL, "Max memory must be >= 0");
```

Comment:
maxmemory = 0 does not make sense, does it?

Reply:
with 0 = force writing to disk? No, you want the request payload in RAM for fast access.
with 0 = unlimited? No ... there really should be a limit. For ownCloud it is currently 10MB when uploading chunks. But admins might change that to improve throughput. Our sync client could automatically adjusts the chunk size based on bandwidth. Unlimited makes no sense.

Reply:
IMO `if(maxmemory <= 0) max_memory = SAPI_POST_BLOCK_SIZE;` should it be, because buffers of a couple of bytes don't really make sense.

Reply:
@weltling that will hide the error message. Personally I prefer to fail Hard and Early. Dunno what the policy is for php.

Reply:
Well, with zero maxmemory it'd effectively enforce HD usage. With maxmemory of a couple of bytes, i'd not see a real use case, too. Both cases are questionable. Thus, if it's less than SAPI_POST_BLOCK_SIZE, reset to SAPI_POST_BLOCK_SIZE and optionally give a warning that the size was automatically increased to a sensible value. About error throwing - it's not done in any part of streams, in the core in general not by default.

Thanks.